src/plugins/gs-plugin-external-appstream.c

```diff
@@ -181,6 +182,11 @@ gs_plugin_external_appstream_refresh_url (GsPlugin *plugin,
 		return FALSE;
 	}
 
+	/* close the file */
+	if (!g_output_stream_close (outstream, cancellable, error)) {
```

Comment:
I would close this in any event right after the call to `g_output_stream_write_all()`, even when that fails, so that we can prevent `g_file_delete()` failing because of dangling reference.